web/src/main/java/org/mskcc/oncotree/api/TumorTypesApi.java

```diff
@@ -3,170 +3,90 @@
-//    @io.swagger.annotations.ApiResponses(value = {
-//        @io.swagger.annotations.ApiResponse(code = 200, message = "OK")})
-//    @RequestMapping(value = "/{id}",
-//        produces = {"application/json"},
-//        method = RequestMethod.GET)
-    public ResponseEntity<TumorTypeResp> tumorTypesIdGet(
-        @ApiParam(value = "The numerical ID of the desired tumor type", required = true)
-        @PathVariable("id") String id,
-        @ApiParam(value = "The callback function name. This has to be used with dataType JSONP.")
-        @RequestParam(value = "callback", required = false) String callback
-
-
-    )
-        throws NotFoundException {
-        // do some magic!
-        return new ResponseEntity<TumorTypeResp>(HttpStatus.OK);
+        Version v = (version == null) ? VersionUtil.getDefaultVersion() : VersionUtil.getVersion(version);
+        Map tumorTypes = CacheUtil.getTumorTypesByVersion(v);
+        Set tumorTypesSet = TumorTypesUtil.flattenTumorTypes(tumorTypes, null);
+        logger.debug("tumorTypesGetFlat() -- returning " + tumorTypesSet.size() + " tumor types");
+        return tumorTypesSet;
     }
 
-
-    @ApiOperation(value = "Tumor Types", notes = "...", response = SearchTumorTypesResp.class)
-    @io.swagger.annotations.ApiResponses(value = {
-        @io.swagger.annotations.ApiResponse(code = 200, message = "An array of tumor types")})
+    @ApiOperation(value = "Tumor Types", notes = "...", response = TumorType.class)
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "An array of tumor types")})
     @RequestMapping(value = "/search",
-        produces = {"application/json"},
+        produces = {APPLICATION_JSON_VALUE},
         method = RequestMethod.POST)
-    public ResponseEntity<SearchTumorTypesPostResp> tumorTypesSearchPost(
+    public List<List<TumorType>> tumorTypesSearchPost(
 
         @ApiParam(value = "queries", required = true) @RequestBody TumorTypeQueries queries
-    )
-        throws NotFoundException {
-        SearchTumorTypesPostResp resp = new SearchTumorTypesPostResp();
-        resp.setMeta(new Meta() {{
-            setCode(200);
-        }});
+    ) {
 
         Version v = queries.getVersion() != null ? VersionUtil.getVersion(queries.getVersion()) : VersionUtil.getDefaultVersion();
         List<List<TumorType>> tumorTypes = new ArrayList<>();
 
         // Cache in tumor types in case no data present
-        CacheUtil.getOrResetTumorTypesByVersion(v);
+        CacheUtil.getTumorTypesByVersion(v);
 
+        // each query has a list of results
         for (TumorTypeQuery query : queries.getQueries()) {
             List<TumorType> matchedTumorTypes = new ArrayList<>();
             matchedTumorTypes = v == null ? new ArrayList<TumorType>() : TumorTypesUtil.findTumorTypesByVersion(query.getType(), query.getQuery(), query.getExactMatch(), v, false);
+            // TODO add back?
```

Comment:
We should either delete this, or add it back and fix it.

Reply:
I think I'm in favor of the levels argument .. but see my large comment below (related to the flat/nonflat returns)